Is this compatible with macOS?

Thank you very much for creating this action!

I'm running to run it on macOS, and I get this error:

```
  ##[error]ENOENT: no such file or directory, open '/etc/lsb-release'
  ##[error]Node run failed with exit code 1
```

Is anyone else experiencing this?

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -11,7 +11,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-16.04, ubuntu-latest]
+        os: [ubuntu-16.04, ubuntu-latest, macos-latest]
     runs-on: ${{ matrix.os }}
     steps:
     - uses: actions/checkout@v1

--- a/src/index.js
+++ b/src/index.js
@@ -1,30 +1,25 @@
 const core = require('@actions/core');
 const exec = require('@actions/exec');
 const io = require('@actions/io');
-const os = require('./os')
+const platform = require('./platform')
 
-function _dependenciesForPlatform(platform) {
-  return {
-    'ubuntu-16.04': 'autoconf bison build-essential libssl-dev libyaml-dev libreadline6-dev zlib1g-dev libncurses5-dev libffi-dev libgdbm3 libgdbm-dev',
-    'ubuntu-18.04': 'autoconf bison build-essential libssl-dev libyaml-dev libreadline6-dev zlib1g-dev libncurses5-dev libffi-dev libgdbm5 libgdbm-dev'
-  }[platform]
-}
+function _dependenciesCmdForPlatformVersion(platform, version) {
+  if (platform == 'Ubuntu') {
+    const dependencies = {
+      '16.04': 'autoconf bison build-essential libssl-dev libyaml-dev libreadline6-dev zlib1g-dev libncurses5-dev libffi-dev libgdbm3 libgdbm-dev',
+      '18.04': 'autoconf bison build-essential libssl-dev libyaml-dev libreadline6-dev zlib1g-dev libncurses5-dev libffi-dev libgdbm5 libgdbm-dev'
+    }[version]
 
-function _getPlatform() {
-  const releaseInfo = os.releaseInfo()
-
-  return `${releaseInfo.distribId.toLowerCase()}-${releaseInfo.distribRelease}`
+    return `sudo apt-get -qq install ${dependencies}`
+  }
 }
 
 async function _installDependencies() {
-  const platform = _getPlatform()
-  const dependencies = _dependenciesForPlatform(platform)
-  if (!dependencies) {
-    core.setFailed(`Cannot find dependencies for platform ${platform}`)
-    return
-  }
+  const dependenciesCmd = _dependenciesCmdForPlatformVersion(platform.type(), platform.version())
 
-  await exec.exec(`sudo apt-get -qq install ${dependencies}`)
+  if (dependenciesCmd) {
+    await exec.exec(dependenciesCmd)
+  }
 }
 
 async function _installRubyBuild() {

--- a/src/platform.js
+++ b/src/platform.js
@@ -1,9 +1,10 @@
 const fs = require('fs')
+const os = require('os')
 const camelCase = require('camelcase')
 
 const LINE_MATCH = /^(\w+)=["']?([\w\s\.]+)["']?$/
 
-export function releaseInfo() {
+function _lsbRelease() {
   const content = fs.readFileSync('/etc/lsb-release', 'utf8')
 
   const obj = {}
@@ -17,4 +18,23 @@ export function releaseInfo() {
   })
 
   return obj
+}
+
+export function type() {
+  const type = os.type()
+  if (type == 'Linux') {
+    try {
+      const lsbRelease = _lsbRelease()
+      return lsbRelease.distribId
+    }
+    catch { }
+  }
+
+  return type
+}
+
+export function version() {
+  if (type() == 'Ubuntu') {
+    return _lsbRelease().distribRelease
+  }
 }

